bot.ml

```diff
@@ -786,15 +786,119 @@ let callback _conn req body =
+              (*let pr =
+                  Option.value_exn ~message:"Attempting to merge an issue."
+                    comment_info.pull_request
+                in
+                GitHub_queries.get_merge_pull_request_refs ~bot_info
+                  ~owner:pr.issue.issue.owner ~repo:pr.issue.issue.repo
+                  ~number:pr.issue.issue.number*)
+              let pr = comment_info.issue in
+              GitHub_queries.get_merge_pull_request_refs ~bot_info
+                ~owner:pr.issue.owner ~repo:pr.issue.repo
+                ~number:pr.issue.number
+              >>= function
+              | Ok merge_info -> (
+                  if
+                    List.exists merge_info.labels ~f:(fun label ->
+                        string_match ~regexp:"needs:.*" label)
+                  then
+                    GitHub_mutations.post_comment ~bot_info
+                      ~message:
+                        (f "@%s: Needs label found, merge aborted."
+                           comment_info.author)
+                      ~id:pr.id
+                  else if
+                    not
+                      (List.exists merge_info.labels ~f:(fun label ->
+                           string_match ~regexp:"kind:.*" label))
+                  then
+                    GitHub_mutations.post_comment ~bot_info
+                      ~message:
+                        (f "@%s: No kind label found, merge aborted."
+                           comment_info.author)
+                      ~id:pr.id
+                    (* merge_info already contains a valid milestone, no need to check if there's one *)
+                    (*else if
+                        List.exists merge_info.reviews ~f:(fun (_, state) ->
+                            match state with
+                            | GitHub_subscriptions.(CHANGES_REQUESTED | DISMISSED)
+                              ->
+                                true
+                            | _ ->
+                                false)
+                      then Lwt_io.print "Changes requested, abort merge\n"*)
+                  else if
+                    not
+                      (List.exists merge_info.reviews ~f:(fun (_, state) ->
+                           match state with
+                           | GitHub_subscriptions.APPROVED ->
+                               true
+                           | _ ->
+                               false))
+                  then
+                    GitHub_mutations.post_comment ~bot_info
+                      ~message:
+                        (f "@%s: PR cannot be merged as it isn't approved yet."
+                           comment_info.author)
+                      ~id:pr.id
+                  else if
+                    not
+                      (List.exists merge_info.assignees ~f:(fun login ->
+                           String.equal login comment_info.author))
+                    (* PR author must be among the assignees *)
```

Comment:
It's the author of the comment requesting a merge who should be among the assignees (in general the only assignee). It looks like the code is correct but the comment must be fixed.